### PR TITLE
fix(reparent) static reparent keeps the original in place hidden

### PR DESCRIPTION
**Pull request by enidemi**
**Problem:**
When reparenting a static element the parent size might change shifting possible reparent targets under the cursor, makes it difficult to target these elements.

**Fix:**
Reparent strategies create a duplicate element in place of the original one and they appear as hidden elements on the canvas. Only if the original elements was not absolute.

**Commit Details:**
- add duplicate element command for non-absolute reparent targets
- these elements are added to `hiddenInstances` and they are hidden in the navigator too
- fix reparent strategies to correctly update the duplicate element uid
- test to check if flex parent is not collapsed during reparent

Before https://utopia.pizza/p/97fce0d6-glory-xylocarp
After https://utopia.pizza/p/97fce0d6-glory-xylocarp/?branch_name=fix-reparent-flex-layout-collapse

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-as-static-strategy.tsx [hunk 1]
```diff
@@ -186,24 +190,28 @@ function applyStaticReparent(
 
             const commandsAfterReorder = [
               ...propertyChangeCommands,
-              setElementsToRerenderCommand([target, newPath]),
+              setElementsToRerenderCommand([target, newPath]), // TODO THIS LIST IS INCOMPLETE, KEEPS GHOSTS IN THE NAVIGATOR
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-as-static-strategy.tsx:193]: I believe this is a bug with all of our existing re-parent strategies

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts [hunk 2]
```diff
@@ -162,3 +174,44 @@ export function cursorForMissingReparentedItems(
...
+      duplicatedElementNewUids[selectedElementString] = newUid
+
+      const newPath = EP.appendToPath(EP.parentPath(elementPath), newUid)
+
+      return [
+        duplicateElement('mid-interaction', elementPath, newUid),
+        wildcardPatch('mid-interaction', {
+          hiddenInstances: { $push: [newPath] },
+        }),
+        hideInNavigatorCommand([newPath]),
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts:210]: So I guess we should use something like this for other re-parenting strategies (but using the old path in those)?

**Review by Rheeseyb — APPROVED** [review]

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts [hunk 3]
```diff
@@ -162,3 +174,44 @@ export function cursorForMissingReparentedItems(
...
+  let duplicatedElementNewUids: { [elementPath: string]: string } = {
+    ...customStrategyState.duplicatedElementNewUids,
+  }
+  const commands = filteredSelectedElements.flatMap((elementPath) => {
+    const element = MetadataUtils.findElementByElementPath(
+      canvasState.startingMetadata,
+      elementPath,
+    )
+    // we want to keep a placeholder element where the original dragged element was to avoid the new parent shifting around on the screen
+    const hasCommonAncestor =
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/reparent-utils.ts:193]: I think it's worth adding a comment here that THIS is the entry point if we want a more finegrained heuristic for happy paths where we don't need the layout-keeping

**Review by balazsbajorics — APPROVED** [review]
nice!

**Pull request merged**